src/operator/rnn-inl.h

```diff
@@ -123,7 +123,7 @@ struct RNNParam : public dmlc::Parameter<RNNParam> {
 };
 
 inline int GetRnnParamSize(int num_layer,
-                           int input_size,
+                           index_t input_size,
```

Comment:
Are you sure you don't have to change size1, size2, proj_size and param_size:
https://github.com/apache/incubator-mxnet/pull/17632/files#diff-6dfdca409e69cc495f286170fe1e553eR143
-
https://github.com/apache/incubator-mxnet/pull/17632/files#diff-6dfdca409e69cc495f286170fe1e553eR152

Reply:
Good point, fixing.

Reply:
sry for changing my opinion again. Since its input size 'size_t' would be better here

Reply:
size_t ? make sure API signature doesn't chage. If thats the case then keep it index_t